src/api.js

```diff
@@ -433,3 +433,50 @@ api.declare({
   }
   return res.status(404).send();
 });
+
+api.declare({
+  name: 'createStatus',
+  title: 'Post a status against a given changeset',
+  description: [
+    'For a given changeset (SHA) of a repository, this will attach a "commit status"',
+    'on github. These statuses are links displayed next to each revision.',
+    'The status is either OK (green check) or FAILURE (red cross), ',
+    'made of a custom title and link.',
+  ].join('\n'),
+  stability: 'experimental',
+  method: 'post',
+  // route and query matches github API
+  // https://developer.github.com/v3/repos/statuses/#create-a-status
+  route: '/repository/:owner/:repo/statuses/:sha',
+  input: 'create-status.json',
+  scopes: [['github:create-status:<owner>/<repo>/<sha>']],
```

Comment:
Ok. So using `deferAuth: true` and `req.satisfies` works fine.
But the issue is that GitHub API doesn't allow getting the branch name of a given commit :/
http://stackoverflow.com/questions/29992441/github-api-identify-the-branches-that-contain-a-given-commit
I also looked at all available `Repositories` API from GitHub:
https://developer.github.com/v3/repos/
As well as Git Data:
https://developer.github.com/v3/git/
But haven't found any that exposes the branch.

I thought we could assert Task branch we are running but that's not what we want to assert, it would still allow passing any SHA from any branch/repo.

The other assertion I could think about would be the SHA for "push" tasks and the "Commit range" for "pull request" tasks. Do we have this information somewhere? Having said that I'm not sure it will be easy to match commit ranges from a scope string.

Note that the scope definition should be easier for Pull request comments API as we would have something like `github:update-status:<org>/<repo>/<PR#ID>`.

Otherwise, is this function the one to modify to correctly flag the tasks with the new necessary scopes?
https://github.com/taskcluster/taskcluster-github/blob/master/src/intree.js#L14
Or does it lives somewhere else in another project?